### DISCUSSION TITLE
[Docs]: KRaft mode not production ready description

### DISCUSSION THREAD

**Discussion by woodliu** [strimzi/strimzi-kafka-operator#9729] [General]
### Suggestion / Problem

In document _7.4. Kafka node pools configuration_ there is a alert as **KRaft mode is not ready for production in Apache Kafka or in Strimzi.** , but  according to [this](https://developer.confluent.io/learn/kraft/#try-it) and [this](https://cwiki.apache.org/confluence/display/KAFKA/KIP-833%3A+Mark+KRaft+as+Production+Ready) document, the kRaft mode has beed production ready from kafka 3.3.

Should the description be changed in Strimzi docuement?

### Documentation Link

[7.4. Kafka node pools configuration](https://strimzi.io/docs/operators/latest/overview#configuration-points-node_pools_str)

**Comment by scholzj** [comment]
I guess we have different requirements for something being production-ready - such as having all key features implemented, having a reasonable quality, being well tested etc.

**Reply by woodliu** [reply]
What confuse me is that, Strimzi is a tool to run Apache Kafka, and kraft is an internal implement of Apache Kafka, i think if the broker can communicate with each other, then the KRaft can work.
Does Strimzi have different implement for KRaft?

**Reply by ppatierno** [reply]
No. Strimzi doesn't have any specific implementation of KRaft even because the operator uses the upstream community Apache Kafka release for the container image. My view is that KRaft was "defined" as production ready too early from the Kafka community when some features were still missing (see support for JBOD which is coming with 3.7.0). There are also a lot of tests missing and documentation holes (my own experience with supporting ZooKeeper to KRaft migration in Strimzi was not that good). Maybe this is what Jakub was referring to. Just having a different idea (compared to Kafka community members) about what can be defined production ready or not.

**Reply by scholzj** [reply]
@woodliu Just because we rely on Apache Kafka, it does not mean that we have to recommend our users to use something that we don't think is ready yet.

**Reply by woodliu** [reply]
@scholzj @ppatierno Thank you very much for the answers. I am looking forward to use the KRaft, but It may take some time to make this feature more mature.
